cliUsage.html

```diff
@@ -0,0 +1,165 @@
+        </p>
+
+        <h2>
+            <a id="examples" class="anchor" href="#examples" aria-hidden="true">
+                <span aria-hidden="true" class="octicon octicon-link"></span>
+            </a>Examples
+        </h2>
+
+        <p>Typical examples of usage Eiffel REMReM Publish CLI are described below.</p>
+
+        <h5>Publish on a given host, given exchange, given domain and given user domain suffix:</h5>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb101 -domain eiffel001 -ud fem001 -mp eiffelsemantics</code></pre>
+
+        <h5>If you want to have the message non persistent add np flag:</h5>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -np -mp eiffelsemantics</code></pre>
+
+        <h5>For loading protocol jars other than eiffelsemantics:</h5>
+        <pre><code>java -Djava.ext.dirs="/path/to/jars/" -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp protocolType</code></pre>
+
+        <p><strong>NOTE:</strong> in the above example, protocol jar file must be present inside "/path/to/jars/" folder.</p>
+
+        <h5>If you want to have the message publishing logs add d flag:</h5>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp eiffelsemantics -d</code></pre>
+
+        <h5>If you want to have the message publishing on RabbitMQ secured port (5671):</h5>
+        <ul>
+            <li>tls option is used to run REMReM on RabbitMQ secured port (5671).</li>
+            <li>tls option value is either 'default' or &lt;version&gt; for secured port.</li>
+        </ul>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp eiffelsemantics -tls 1.2</code></pre>
+
+        <p><strong>NOTE:</strong> for RabbitMQ default port 5672, no need to pass -tls option.</p>
+
+        <h5>If you want to have the message publishing on RabbitMQ with given routing key:</h5>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp eiffelsemantics -rk myroutingkey</code></pre>
+
+        <h5>If you want to have the message publishing on RabbitMQ with given tag:</h5>
+        <pre><code>java -jar publish-cli.jar -f publishMessages.json -en rabbit -mb mb123 -mp eiffelsemantics -tag production</code></pre>
+
+        <p>Typical output for these examples is:</p>
+<pre>[
+  {
+    "id": "963da060-f2cf-4370-a68d-67fd872def36",
+    "status_code": 200,
+    "result": "SUCCESS",
+    "message": "Event sent successfully"
+  }
+]</pre>
+
+        <h2>
+            <a id="statusCodes" class="anchor" href="#statusCodes" aria-hidden="true">
+                <span aria-hidden="true" class="octicon octicon-link"></span>
+            </a>Status Codes
+        </h2>
+
+        <p>For each user request Eiffel REMReM Publish generate response in JSON with internal status code and message.</p>
```

Comment:
Internal status code and message is not applicable for CLI. Better to link to the java file for exit codes.

https://github.com/Ericsson/eiffel-remrem-publish/blob/master/publish-cli/src/main/java/com/ericsson/eiffel/remrem/publish/cli/CLIExitCodes.java

Reply:
I tested cli one more time right now and I got status codes and messages as a response

Reply:
I can guarantee you that if cli fails internally before trying to publish the message, it will exit with a code from the above mentioned file. The other HTTP status codes are returned from publishing steps where an HTTP request will be performed. This is important to be mentioned if someone will write scripts to use the cli.